components/Header.jsx

```diff
@@ -0,0 +1,303 @@
+import Link from "next/link"
+import { FormattedMessage } from "react-intl"
+
+import mastodonLogo from "../public/logos/logo-full-purple.svg"
+import Image from "next/image"
+import { useState, useEffect, useRef } from "react"
+import classNames from "classnames"
+import { locales } from "../data/locales"
+import MenuToggle from "./MenuToggle"
+import SVG from "react-inlinesvg"
+
+const useMenu = ({ navigationItems }) => {
+  const [mobileMenuOpen, setMobileMenuOpen] = useState(false)
+  const [primaryMenuItemIndex, setPrimaryMenuItemIndex] = useState(0)
+  const [secondaryMenuItemIndex, setSecondaryMenuItemIndex] = useState(null)
+  const [menuBarHasFocus, setMenuBarHasFocus] = useState(false)
+  const rootElement = useRef(null)
+
+  // Navigation
+  const navigateHorizontally = (direction) => {
+    setPrimaryMenuItemIndex(
+      (primaryMenuItemIndex + direction + navigationItems.length) %
+        navigationItems.length
+    )
+    setSecondaryMenuItemIndex(null)
+  }
+  const navigateVertically = (direction) => {
+    const secondaryItems = navigationItems[primaryMenuItemIndex].childItems
+    if (secondaryItems) {
+      if (secondaryMenuItemIndex === null) {
+        setSecondaryMenuItemIndex(
+          direction === 1 ? 0 : secondaryItems.length - 1
+        )
+      } else {
+        setSecondaryMenuItemIndex(
+          ((secondaryMenuItemIndex || 0) + direction + secondaryItems.length) %
+            secondaryItems.length
+        )
+      }
```

Comment:
I'd maybe add some comments here....maybe before line 30 like `Handles selecting the first/last item when the dropdown is closed` and before line 35 `Handles selecting the next/previous item when the dropdown is open`. Just to help readability for the logic inside `setSecondaryMenuItemIndex()`